### PR TITLE
[Enhancement] view check modified columns

### PR DESCRIPTION
**Pull request by Seaven**
Fixes #issue

* Add status on View and change it
when:
  * execute `ALTER COLUMN`
  * execute `DROP TABLE`
  * execute `SWAP TABLE`
* Fix semi-type error in view 

example:
```
MySQL test_db> create view xxx as select * from t0;
Query OK, 0 rows affected
Time: 0.021s
MySQL test_db> alter table t0 MODIFY COLUMN v3 largeint;
Query OK, 0 rows affected
Time: 4.060s
MySQL test_db> select * from xxx;
(1064, 'Getting analyzing error. Detail message: The type of column[v3] on View[`test_db`.`xxx`] is different with query, please check the source table has been modified.')
MySQL test_db>
```

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 39 / 51 (76.47%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/View.java|5|9|55.56%|[170, 171, 172, 179]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|20|27|74.07%|[779, 786, 787, 788, 789, 791, 793]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|14|15|93.33%|[667]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 52 / 59 (88.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/View.java|5|9|55.56%|[170, 171, 172, 179]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|14|16|87.50%|[233, 242]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|14|15|93.33%|[667]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|16|16|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 52 / 59 (88.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/View.java|5|9|55.56%|[170, 171, 172, 179]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|14|16|87.50%|[233, 242]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|14|15|93.33%|[667]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|16|16|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 52 / 59 (88.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/View.java|5|9|55.56%|[170, 171, 172, 179]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|14|16|87.50%|[233, 242]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|14|15|93.33%|[667]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|16|16|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 73 / 81 (90.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/View.java|5|9|55.56%|[170, 171, 172, 179]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|14|16|87.50%|[233, 242]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|14|15|93.33%|[664]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToViewBuilder.java|20|21|95.24%|[24]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|16|16|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java [hunk 1]
```diff
@@ -688,6 +692,26 @@ public Scope visitView(ViewRelation node, Scope scope) {
                 fields.add(field);
             }
 
+            // check view schema
+            Map<String, Column> columns =
+                    node.getView().getColumns().stream().collect(Collectors.toMap(Column::getName, c -> c));
+
+            for (Field field : fields) {
+                if (!columns.containsKey(field.getName())) {
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:700]: column need case insensitive.

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToViewBuilder.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public static class AST2ViewBuilderVisitor extends AstToSQLBuilder.AST2SQLBuilderVisitor {
+        public AST2ViewBuilderVisitor(boolean simple, boolean withoutTbl) {
+            super(simple, withoutTbl);
+        }
+
+        @Override
+        public String visitArrayExpr(ArrayExpr node, Void context) {
+            StringBuilder sb = new StringBuilder();
```
[the remark this answers is not in the document]
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToViewBuilder.java:36]: just think `toSQL` don't need type info, but you're right, it's unnecessary now

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 73 / 81 (90.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/View.java|5|9|55.56%|[170, 171, 172, 179]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|14|16|87.50%|[233, 242]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|14|15|93.33%|[664]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToViewBuilder.java|20|21|95.24%|[24]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|16|16|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 70 / 77 (90.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/View.java|5|9|55.56%|[170, 171, 172, 179]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|14|16|87.50%|[233, 242]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|15|16|93.75%|[664]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToSQLBuilder.java|17|17|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|16|16|100.00%|[]|

**Review by murphyatwork — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/View.java [hunk 3]
```diff
@@ -62,6 +62,9 @@
 public class View extends Table {
     private static final Logger LOG = LogManager.getLogger(GlobalStateMgr.class);
 
+    public static final int VIEW_STATUS_NORMAL = 0;
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/catalog/View.java:65]: please use enum. why don't have the inactive status?

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 73 / 80 (91.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/View.java|8|12|66.67%|[172, 173, 174, 181]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|14|16|87.50%|[233, 242]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|15|16|93.75%|[664]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToSQLBuilder.java|17|17|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|16|16|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Pull request merged**